grow/default.nix

```diff
@@ -108,12 +108,15 @@
             if l.typeOf block == "set"
             then block
             else block signature;
-          imported =
-            if isFile
-            then Target' oPath.file' (import' oPath.file' oPath.file)
-            else if isDir
-            then Target' oPath.dir' (import' oPath.dir' oPath.dir)
-            else throw "unreachable!";
+          imported = let
+            block =
+              if isFile
+              then oPath.file' (import' oPath.file' oPath.file)
+              else if isDir
+              then oPath.dir' (import' oPath.dir' oPath.dir)
+              else throw "unreachable!";
+          in
+            Target' (cellBlock.__type or null) block;
```

Comment:
In my view, this change would bring about the following inconsistency:
- Block Types are to be said "types"
- Only to be potentially superseded / complemented  or `worst, contradicted by `__type`

Reply:
Well I think thats still better than user putting some non-sensical target into a blocktype and having absolutely no idea why nothing works or something broke.

That happened a few times at work already. All targets should more or less be the same type right? I mean in cases like the `data` type that are a bit more liberal, you can just stick with an `any`.

We could call it `targetType` if that makes it clearer.

> further design on how it relates to shared types (Block Types)

I think allowing blocktypes to contain all sorts of types of targets is probably an overcomplication and a code smell and you should probably redesign your blocktype to be more focused. But that's just my opinion.

Reply:
> Well I think thats still better than user putting some non-sensical target

No, we know it's not well designed.

Therefore, we should spend some cycles on design.

Paisano needs to be able to suffer temporary tension between (the time it takes for) good design and user pains.

Reply:
> I think allowing blocktypes to contain all sorts of types of targets is probably an overcomplication and a code smell and you should probably redesign your blocktype to be more focused. 

There you have it. :smile: That would be a better design.

Reply:
> No, we know it's not well designed.

I guess I wasn't ware of that, can you expand?

> There you have it.  That would be a better design.

Now I am really confused, that is my design here, far as I can tell.

Reply:
Ah! my bad, how can I undo this conversation? :smile: 

I turned left instead of right when analyzing. :facepalm: